Loop/Views/Presets/EditPresetView.swift

```diff
@@ -135,136 +62,92 @@ struct EditPresetView: View {
+                            .multilineTextAlignment(.trailing)
+                            .focused($isTextFieldFocused)
+                            .foregroundColor(.secondary)
+                    } else {
+                        Text(preset.name)
+                            .foregroundColor(.secondary)
+                    }
                 }
-                .padding(12)
-                .background(color.opacity(0.1))
-                .cornerRadius(12)
             }
-        }
-    }
 
-
-    var correctionSection: some View {
-        CompactSection {
-            Button {
-                navigateToCorrectionRangeEditor = true;
-            } label: {
-                VStack(alignment: .center, spacing: 12) {
-                    HStack {
-                        Text("Correction Range")
-                            .font(.system(size: 17, weight: .semibold))
-                        Spacer()
-                        Image(systemName: "chevron.right")
-                            .foregroundColor(.secondary)
-                    }.padding(.bottom, 10)
-                    VStack(spacing: 4) {
-                        if let range = preset.correctionRange {
-                            correctionRangeLabel(range: range)
-                            Text("Adjusted Range")
-                        } else {
-                            correctionRangeLabel(range: scheduledRange)
-                            Text("Scheduled Range")
+            CardSection(
+                content: {
+                    Button(action: {
+                        showingPicker = true
+                    }) {
+                        HStack {
+                            Text("Duration")
+                                .foregroundColor(.primary)
+                            Spacer()
+                            Text(preset.duration.localizedTitle)
+                                .foregroundColor(.secondary)
+                            if preset.canAdjustDuration {
+                                Image(systemName: "chevron.right")
+                                    .foregroundColor(.gray)
+                            }
                         }
-                    }
-                    guardrailWarningIfNecessary
-                        .font(.system(size: 13, weight: .semibold))
-                        .foregroundColor(.primary)
-                        .padding(.bottom, 5)
-                        .padding(.horizontal, 2)
+                    }.disabled(!preset.canAdjustDuration)
+                },
+                footerText: preset.canAdjustDuration ? nil : "Duration and Name not configurable for this preset."
+            )
+        }
+        .sheet(isPresented: $showingPicker) {
+            VStack(alignment: .center, spacing: 24) {
+                HStack {
+                    Text("Duration")
+                        .font(.system(size: 17, weight: .regular))
```

Comment:
This is just the default `.body` font but doesn't scale. Is that intended?

Reply:
Removed

Reply:
cleaned up